geode-core/src/distributedTest/java/org/apache/geode/management/JMXMBeanReconnectDUnitTest.java

```diff
@@ -156,47 +341,51 @@ public void testLocalBeans_MaintainLocatorAndCrashServer() {
    * All MBeans not related to the killed member should remain the same when a member is killed.
    */
   @Test
-  public void testRemoteBeanKnowledge_MaintainServerAndCrashLocator() throws IOException {
-    // check that the initial state is good
-    List<ObjectName> initialL1Beans = jmxConToLocator1.getGemfireFederatedBeans();
-    List<ObjectName> initialL2Beans = jmxConToLocator2.getGemfireFederatedBeans();
-    assertThat(initialL1Beans).containsExactlyElementsOf(initialL2Beans).hasSize(NUM_REMOTE_BEANS);
-
-    // calculate the expected list for use once the locator has crashed
-    List<ObjectName> expectedIntermediateBeanList = initialL1Beans.stream()
-        .filter(excludingBeansFor("locator-0")).collect(toList());
-
-    // crash the locator
-    locator1.forceDisconnect(TIMEOUT, TimeUnit.MILLISECONDS, RECONNECT_MAILBOX);
-
-    // wait for the locator's crash to federate to the remaining locator
-    List<ObjectName> intermediateL2Beans = new ArrayList<>();
-    await().untilAsserted(() -> {
-      intermediateL2Beans.clear();
-      intermediateL2Beans.addAll(jmxConToLocator2.getGemfireFederatedBeans());
-
-      assertThat(intermediateL2Beans)
-          .containsExactlyElementsOf(expectedIntermediateBeanList)
-          .hasSameSizeAs(expectedIntermediateBeanList);
+  public void locatorMXBeansOnOtherLocatorAreRestoredAfterCrashedLocatorReturns() {
+    locator1VM.invoke(() -> {
+      RECONNECT_LATCH.set(new CountDownLatch(1));
+
+      addReconnectListener(new ReconnectListener() {
```

Comment:
@igodwin Here's one of the implementations of ReconnectListener in the new test which doesn't implement both methods. If we remove `default`, then this impl would have to provide an empty impl of `onReconnect`.